### PR TITLE
remove `proxy` from trpc/next

### LINKED ISSUE #2607: [v10] feat: update `setupNext` to not include `proxy`
### Describe the feature you'd like to request

Right now we do 

```ts

export const trpc = setupTRPC<AppRouter>({
  config({ ctx }) {
    // [..]
  })
});
```

and we call it we use 

```ts
trpc.withTRPC(router)`
trpc.proxy.hello.useQuery(/* ... /*)
```

### Describe the solution you'd like to see

Change this to 

```ts

export const trpc = setupTRPC<AppRouter>({
  config({ ctx }) {
    // [..]
  })
});
```

and we call it we use 

```ts
trpc.withTRPC(router)`
trpc.hello.useQuery(/* ... /*)
```

(no `.proxy`)

### Desribe alternate solutions

n/a

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by juliusmarminge**
Closes #2607

## 🎯 Changes

Removes the need to prepend procedures with `.proxy` when using @trpc/next.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: examples/next-minimal-starter/src/pages/index.tsx [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import { trpc } from '../utils/trpc';
 
 export default function IndexPage() {
-  const hello = trpc.proxy.hello.useQuery({ text: 'client' });
+  const hello = trpc.hello.useQuery({ text: 'client' });
+  const utils = trpc.useUtilsContext();
+  utils.hello.invalidate();
```
Comment (KATT) [inline examples/next-minimal-starter/src/pages/index.tsx:6]:
```suggestion
```

File: packages/react/src/shared/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export * from '../internals/getQueryKey';
+
+export {
+  createReactQueryUtilsProxy,
+  type DecoratedProcedureUtilsRecord,
+} from '../internals/utilsProxy';
```
Comment (juliusmarminge) [inline packages/react/src/shared/index.ts:6]: should we move these stuff out somewhere or just mark them `@internal` now that they can be imported elsewhere?

Comment (KATT) [reply packages/react/src/shared/index.ts:6]: Yeah, that'd be nice 👌

**Review by KATT — APPROVED** [review]

File: packages/react/src/shared/index.ts (hunk 2 above)
Comment (KATT) [reply packages/react/src/shared/index.ts:6]: You can just add a `FIXME` comment too :D

**Pull request merged**